CMakeLists.txt

```diff
@@ -267,16 +267,23 @@ endif()
 # Make sure the generated paths exist
 execute_process(COMMAND mkdir -p "${CMAKE_BINARY_DIR}/generated")
 
-# We need to link some packages as dynamic/dependent.
-set(CMAKE_FIND_LIBRARY_SUFFIXES .dylib .so)
-find_package(OpenSSL REQUIRED)
-find_package(BZip2 REQUIRED)
-find_package(Dl REQUIRED)
-find_package(Readline REQUIRED)
+# Apple has deprecated OpenSSL, static link with brew bottled libressl
+if(APPLE)
+  set(CMAKE_FIND_LIBRARY_SUFFIXES .a)
```

Comment:
Is `BUILD_LINK_SHARED=1` just about adding the environment variable in the homebrew osquery formula. And if so, will that affect other dependencies like rocksdb? And when should be update the osquery formula, after this is merged or after a release?